extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/sql/resources/SqlStatementResource.java

```diff
@@ -0,0 +1,767 @@
+      ColumnMappings columnMappings = msqControllerTask.getQuerySpec().getColumnMappings();
+      List<SqlTypeName> sqlTypeNames = msqControllerTask.getSqlTypeNames();
+      if (sqlTypeNames == null) {
+        return Optional.empty();
+      }
+      List<ColNameAndType> signature = new ArrayList<>(columnMappings.size());
+      int index = 0;
+      for (String colName : columnMappings.getOutputColumnNames()) {
+        signature.add(new ColNameAndType(colName, sqlTypeNames.get(index).getName()));
+        index++;
+      }
+      return Optional.of(signature);
+    }
+    return Optional.empty();
+  }
+
+
+  private void checkTaskPayloadOrThrow(TaskPayloadResponse taskPayloadResponse, String queryId) throws QueryException
+  {
+    if (taskPayloadResponse == null || taskPayloadResponse.getPayload() == null) {
+      throw new QueryException(QueryException.UNKNOWN_EXCEPTION_ERROR_CODE,
+                               StringUtils.format(
+                                   "Could not get payload details of query[%s] from the overlord",
+                                   queryId
+                               ), null, null
+      );
+    }
+
+    if (MSQControllerTask.class != taskPayloadResponse.getPayload().getClass()) {
+      throw new QueryException(QueryException.UNKNOWN_EXCEPTION_ERROR_CODE,
+                               StringUtils.format(
+                                   "Fetched an unexpected payload details of query[%s] from the overlord.",
+                                   queryId
+                               ), null, null
+      );
+    }
+  }
+
+  private Optional<ResultSetInformation> getSampleResults(
+      String asyncResultId,
+      boolean isSelectQuery,
+      SqlStatementState sqlStatementState
+  )
+  {
+    // only populate sample results in case a select query is successful
+    if (isSelectQuery && sqlStatementState == SqlStatementState.SUCCESS) {
+      Map<String, Object> report = overlordClient.getTaskReport(asyncResultId);
+      Optional<List<Object>> rows = getResults(report);
+
+      if (rows.isPresent()) {
+        return Optional.of(new ResultSetInformation(
+            null,
+            false,
+            (long) rows.get().size(),
+            null,
+            rows.get()
+        ));
+      } else {
+        return Optional.empty();
+      }
+
+    } else {
+      return Optional.empty();
+    }
+  }
+
+
+  private Optional<SqlStatementResult> getStatementStatus(String queryId, String currentUser, boolean withResults)
+      throws QueryException, ForbiddenException
+  {
+    TaskStatusResponse taskResponse = overlordClient.getTaskStatus(queryId);
+    if (taskResponse == null) {
+      return Optional.empty();
+    }
+
+    TaskStatusPlus statusPlus = taskResponse.getStatus();
+    if (statusPlus == null || !MSQControllerTask.TYPE.equals(statusPlus.getType())) {
+      return Optional.empty();
+    }
```

Comment:
is this expected? If not, can we log some warning?

Reply:
I think we should not add a warning since its bad user input.